src/server/server.rs

```diff
@@ -1,30 +1,120 @@
+use std::{
+    collections::HashMap,
+    env,
+    io::Error as IoError,
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+    thread,
+};
+
+pub struct SocketMap {
+    socket_ids: HashMap<u32, HashMap<u32, WebSocket<TcpStream>>>, //first: computation id ; second:party id
+    computation_ids: HashMap<SocketAddr, u32>,
+    //computation_ids: HashMap<WebSocket<TcpStream>, u32>,
+    party_ids: HashMap<SocketAddr, u32>, // party id
+}
 
 pub struct Server {
     pub name: String,
+    //socketMap: SocketMap,
 }
 
 impl Server {
     pub fn on(&self) {
+        let mut socket_map = SocketMap {
+            socket_ids : HashMap::new(),
+            computation_ids : HashMap::new(),
+            party_ids :HashMap::new(),
+        };
+        socket_map.socket_ids.insert(1, HashMap::new());
+        let socket_map = Arc::new(RwLock::new(socket_map));
+
         let server = TcpListener::bind("127.0.0.1:9001").unwrap();
-        for stream in server.incoming() {
+        //let shared_message = Arc::new(Mutex::new(String::from("")));
+        //let websockets_hashmap = Arc::new(RwLock::new(HashMap::new()));
+        let counter = Arc::new(Mutex::new(0));
+        
+
+        while let Ok((stream, addr)) = server.accept() {
+            
+            //let websockets_hashmap = Arc::clone(&websockets_hashmap);
+            let socket_map = Arc::clone(&socket_map);
+            let counter = Arc::clone(&counter);
+
             spawn(move || {
-                let mut websocket = accept(stream.unwrap()).unwrap();
-                loop {
-                    let msg = websocket.read_message().unwrap();
-                    println!("Received: {}", msg);
+                println!("new thread!");
+                let id;
+                {
+                    let mut num = counter.lock().unwrap();
+                    *num += 1;
+                    println!("Received: {}", &num);
+                    id = num.clone();
+                }
+                let websocket = accept(stream).unwrap();
 
-                    // We do not want to send back ping/pong messages.
-                    if msg.is_binary() || msg.is_text() {
-                        websocket
-                            .write_message(Message::Text("Server message".into()))
-                            .unwrap();
-                        //websocket.write_message(msg).unwrap();
-                    }
+                //build SocketMap
+                {
+                    let mut socket_map = socket_map.write().unwrap();
+                    socket_map.computation_ids.insert(addr, 1);
+                    socket_map.party_ids.insert(addr, id);
+                    let mut socket_ids = &mut socket_map.socket_ids;
+                    socket_ids.get_mut(&1).unwrap().insert(id, websocket);
+                    
+                    println!("{:?}", socket_ids.get(&1).unwrap().get(&id));
                 }
+
+                let mut planner = periodic::Planner::new();
```

Comment:
If changing to the async library (tokio-tungstenite) allows us to remove the timer from this code I think it's worth it. Although this works to avoid race-conditions and keep every thread active, using a timer is pretty fragile and will impact performance significantly down the line.

Reply:
maybe there are also minor scoping things we can do to acquire/release locks in each thread.